connector/connect/server/src/main/scala/org/apache/spark/sql/connect/service/SparkConnectService.scala

```diff
@@ -143,7 +170,13 @@ class SparkConnectService(debug: Boolean)
       responseObserver: StreamObserver[proto.ExecutePlanResponse]): Unit = {
     try {
       new SparkConnectStreamHandler(responseObserver).handle(request)
-    } catch handleError("execute", observer = responseObserver)
+    } catch {
+      handleError(
+        "execute",
+        observer = responseObserver,
+        userId = request.getUserContext.getUserId,
+        sessionId = request.getSessionId)
```

Comment:
Is it possible to throw any error when getting the UserContext and SessionId?

Reply:
user context should not be null, userid and session id can be null (but should not). Better safe than sorry is my motto here.

Reply:
Do we need throw exceptions or assert?  Assert might be proper if the `null` stuff is due to developer errors which are expected to not happen and fixed as bug